lib/gruf/instrumentation/request_logging/hook.rb

```diff
@@ -139,6 +141,29 @@ def sanitize(params = {})
         def options
           super().fetch(:request_logging, {})
         end
+
+        def redact!(parts, i, params, redacted_string)
+          return if i == parts.size || !params.key?(parts[i])
+          if i == parts.size - 1
+            if params[parts[i]].is_a? Hash
+              hash_deep_redact!(params[parts[i]], redacted_string)
+            else
+              params[parts[i]] = redacted_string
+            end
+            return
+          end
+          redact!(parts, i + 1, params[parts[i]], redacted_string)
+        end
+
+        def hash_deep_redact!(hash, redacted_string)
```

Comment:
Can you please add docblocks to this method with param/return descriptions and a summary?